frontend/src/store/player-element.ts

```diff
@@ -8,18 +8,26 @@ import JASSUB from 'jassub';
 import jassubWorker from 'jassub/dist/jassub-worker.js?url';
 import jassubWasmUrl from 'jassub/dist/jassub-worker.wasm?url';
 import { computed, nextTick, shallowRef, watch } from 'vue';
-import { playbackManager } from './playback-manager';
+import { playbackManager, type PlaybackExternalTrack } from './playback-manager';
 import { isArray, isNil, sealed } from '@/utils/validation';
 import { mediaElementRef } from '@/store';
 import { CommonStore } from '@/store/super/common-store';
 import { router } from '@/plugins/router';
 import { remote } from '@/plugins/remote';
+import { SubtitleDeliveryMethod } from '@jellyfin/sdk/lib/generated-client/models/subtitle-delivery-method';
+import { isMobile } from '@/utils/browser-detection';
```

Comment:
I also plan to remove this in the migration to MediaCapabilities. Why is needed?
I've explained this in some other issues/PRs but can't find them right now, but basically the reasoning behind it it's that user agents are in its way to deprecation and the barriers of what a mobile device is really unambiguos nowadays? Is a VisionPro a mobile device? A Surface? An Android Auto/CarPlay unit?

We strictly need to check the relevant thing we need (for instance touch support) or size/scaling, nothing else. I aim for this to be as generic (but as precise) as possible in this regard.

Reply:
It's used to check how to apply subtitles, since displaying the custom element isn't always possible on some devices i.e. inside the iOS fullscreen media element (which is it's own custom element). So in cases like that you'd still apply a track to the video element.

However you're correct this is a very generic fix that won't be perfect in all scenarios, so I'll start thinking of another way to implement this.

Reply:
I upstreamed to VueUse all the logic that makes fullscreen work in Safari, take a look: https://github.com/vueuse/vueuse/pull/2822

If I recall correctly, the limitations about custom elements inside video only applied when playing fullscreen. Maybe you could extract this check: https://github.com/jellyfin/jellyfin-vue/blob/master/frontend/src/composables/use-playback.ts#L21

If fullscreeen works without element, then you can assume it's not iOS, if it doesn't, then it's iOS (add proper comment block for that of course for future reference).